t/utils/utils.go

```diff
@@ -95,9 +96,19 @@ func LoadFromFile(filename string) string {
 	return buf.String()
 }
 
+// EnsureNodeSyncInput interface to be used on EnsureNodeSync
+type EnsureNodeSyncInput interface {
+	// NodeConfig returns reference to running node's configuration
+	NodeConfig() (*params.NodeConfig, error)
+	// LightEthereumService exposes reference to LES service running on top of the node
+	LightEthereumService() (*les.LightEthereum, error)
+	// PeerCount returns number of connected peers
+	PeerCount() int
+}
+
 // EnsureNodeSync waits until node synchronzation is done to continue
 // with tests afterwards. Panics in case of an error or a timeout.
-func EnsureNodeSync(nodeManager common.NodeManager) {
+func EnsureNodeSync(nodeManager EnsureNodeSyncInput) {
```

Comment:
I think we should provide `nodeConfig` and `les` here. For the `peersCount`, I don't know, do we really need it here?

Reply:
Well, nothing has changed here, tests referencing this function will still be providing an instance of node.NodeManager. 
The only change here, is Instead of requiring them to accomplish `common.NodeManager` interface they're now accomplishing one much more simple [EnsureNodeSyncInput](https://github.com/status-im/status-go/pull/780/files/20b95c0d1a2b82d5ce16e15f2d1c816b0398e7b3#diff-1f1114839889ebb0318dd76ca2aa5bd0R100) which is already providing access to `les` and `peersCount` through `LightEthereumService() (*les.LightEthereum, error)` and `PeerCount() int` methods

Reply:
That makes sense. But then in the other parts of the code, you call these structures `Provider`s. So let's keep the naming consistent.
And the variable name is still `nodeManager` which isn't true.

Reply:
Oops, thats true, i'll change it now :P